feat: add reserved port validation (#55237)
vercel/next.js: errors/reserved-port.mdx

---
title: Reserved Port
---

## Why This Error Occurred

Server was started on a reserved port. For example, `4045` is reserved for the Network Paging Protocol (npp).

```
next start -p 4045
```

or

```
next dev -p 4045
```

Starting the server on a reserved port will result in an error.

## Possible Ways to Fix It

> Visit the [port blocking](https://fetch.spec.whatwg.org/#port-blocking) section of WHATWG's fetch spec and ensure your port is not on the known reserved port list.

Reviewer: ```suggestion
Change the provided port to ensure it's not listed in the [Port Blocking](https://fetch.spec.whatwg.org/#port-blocking) section of WHATWG's fetch spec.
```
